### DISCUSSION TITLE
The thread and request thread of the middleware are not the same

### DISCUSSION THREAD

**Discussion by houxinlin** [fastapi/fastapi#10434] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
import uvicorn
from fastapi_wsgi_middleware import FastapiWSGIMiddleware
from starlette.middleware.base import BaseHTTPMiddleware
app=FastAPI()
my_middleware = FastapiWSGIMiddleware(app)
app.add_middleware(BaseHTTPMiddleware, dispatch=my_middleware)

@app.get("/")
def test(request: Request):
    current_thread_name = threading.current_thread().name
    print(f"Current thread name: {current_thread_name}")

    return {"message": "Hello, World!"}

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000)



class FastapiWSGIMiddleware(WSGIMiddleware):
    def __init__(self, app):
        super(FastapiWSGIMiddleware, self).__init__(app,None, {
            "name": const.CONTAINER_FASTAPI,
            "version": fastapi.__version__
        })

    async def __call__(self, request: Request, call_next):
        current_thread_name = threading.current_thread().name
        print(f"Current thread name: {current_thread_name}")
        ......
```

### Description

I want them to be on the same thread, how to do it

### Operating System

Linux, Windows, macOS

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.7.7

**Comment by jgould22** [comment, marked as answer]
Using a `def test` instead of `async def test` will tell fast api to spawn that function on a thread pool (with the assumption there is blocking code in the controller)

If there is no blocking code you may be able to asyn def on test but I am not sure.
